src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1846,10 +1857,24 @@ public void execute() throws GitException, InterruptedException {
                 // Split fields into branch name, SHA1, and rest of line
                 // Fields are separated by one or more spaces
                 String[] branchVerboseOutput = line.substring(2).split(" +", 3);
+                if (branchVerboseOutput[1].equals("->")) {
+                    links.add(line);
+                }
                 if (branchVerboseOutput[1].length() == 40) {
                     branches.add(new Branch(branchVerboseOutput[0], ObjectId.fromString(branchVerboseOutput[1])));
                 }
             }
+            for (String link : links) {
```

Comment:
indeed. I had to re-implement getRemoteBranches in plain git cli as the legacy code was relying on JGIT for this CLI implementation (sic), and JGit doesn't behave the same as `git branch -r`.